src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -218,10 +220,12 @@ public void verifyAgencyAccess(final String agencyId) {
         if (agencyIds.isEmpty()) {
             if (authenticationFacade.isClientOnly()) {
                 logClientUnauthorisedAccess(agencyId);
+                throw new AccessDeniedException(format("Client not authorised to access agency with id %s.", agencyId));
```

Comment:
this message seems slightly odd - clients aren't restricted by `agencyId` anyway?

Reply:
If they don't have the override role then the client isn't allowed to access the method if it has VerifyAgencyAccess